### PR TITLE
Refactors #247 - Fixed size logic

### PR DESCRIPTION
**Pull request by dgilperez**
This is a refactor from #247, work in progress. 

The reason for opening a new PR is that #247 implementation is working properly in a production app, while this code is not (although specs are green) and because I think it makes sense to consider both approaches.

### CODE REVIEW

File: lib/counter_culture/counter.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module CounterCulture
   class Counter
-    CONFIG_OPTIONS = [ :column_names, :counter_cache_name, :delta_column, :foreign_key_values, :touch, :delta_magnitude]
+    CONFIG_OPTIONS = [ :column_names, :counter_cache_name, :delta_column, :foreign_key_values, :touch, :delta_magnitude, :condition ]
```
Comment (magnusvk) [inline lib/counter_culture/counter.rb]: Separate question, but is this change unrelated? I'm a little confused as to the purpose of this.

Comment (dgilperez) [reply lib/counter_culture/counter.rb]: You're right, this looks like another unrelated fix, related with the ability of passing `if:` to `:counter_culture`. @WaKeMaTTa can you add a bit more here, please?

Comment (phucbnh-1838) [reply lib/counter_culture/counter.rb]: in my opinion you can write this as follow: `CONFIG_OPTIONS = %i[ column_names counter_cache_name delta_column foreign_key_values touch delta_magnitude condition ]`

**Comment by magnusvk** [comment]
Closing this as it's now old.

**Pull request closed without merging**